### DISCUSSION TITLE
Customizable "layouts/views"

### DISCUSSION THREAD

**Discussion by ChuckJonas** [keystonejs/keystone#6133] [Feature requests]
Currently there is little control over what happens when the user opens a record.  You can define custom views and set "readonly", "hidden", etc but this isn't very flexible.

Some general ideas:

- Support for custom views (via configuration) and some form of "routing" to control which view is show
- support for Non-field components
- Control over order and Layout of fields
- conditional rendering of fields and field modes
- Toggle between View vs Edit (with ability to set default mode)

**Comment by raveling** [comment]
@ChuckJonas thanks for sharing these ideas! 🙌 

Generally speaking, we're aligned with what you've shared and have these ideas listed as part of a holistic plan to improve the Admin UI. This work will begin in earnest once the `General Availability` release for Keystone 6 is complete.
